sentinel-cluster/sentinel-cluster-client-default/src/main/java/com/alibaba/csp/sentinel/cluster/client/codec/data/ParamFlowRequestDataWriter.java

```diff
@@ -55,36 +54,37 @@ public void writeTo(ParamFlowRequestData entity, ByteBuf target) {
         target.writeInt(amount);
 
         // Serialize parameters with type flag.
-        for (Object param : entity.getParams()) {
-            encodeValue(param, target);
+        Object[] array = params.toArray();
+        for (int index = 0; index < amount; index++) {
```

Comment:
There are some problems. Assume the parameter type list is `[double, float, SomePojo, int]`, the customized `SomePojo` object will be ignored so the calculated amount will be 3. Then if we use `[0..2]` as the index to encode values, the third `int` value would be lost. Maybe we need a `List<Object> resolveValidParams(Collection<Object> params)` method (could be refactored from `calculateParamTransportSize` method) , then we could get the amount and all values in order from the method.